locations/spiders/inpost_it.py

```diff
@@ -0,0 +1,110 @@
+import re
+
+from locations.categories import Categories, apply_category
+from locations.hours import CLOSED_IT, DAYS_IT, NAMED_DAY_RANGES_IT, NAMED_TIMES_IT, OpeningHours
+from locations.json_blob_spider import JSONBlobSpider
+
+
+class InpostITSpider(JSONBlobSpider):
+    name = "inpost_it"
+    allowed_domains = ["inpost.it"]
+    start_urls = []  # ["https://inpost.it/sites/default/files/points.json"]
+    locations_key = "items"
+    requires_proxy = True
+
+    operator = {"operator": "InPost", "operator_wikidata": "Q3182097"}
+    attributes = {"parcel_mail_in": "yes", "parcel_pickup": "yes"}
+    brand_locker = {"brand": "InPost", "brand_wikidata": "Q3182097", "nsi_id": "inpost-6b37ec"}
+    brand_partner = {"post_office:brand": "InPost", "post_office:brand:wikidata": "Q3182097"}
+
+    def start_requests(self):
+        for domain in self.allowed_domains:
+            self.start_urls.append(f"https://{domain}/sites/default/files/points.json")
+        yield from super().start_requests()
+
+    def pre_process_data(self, v):
+        # this mapping comes from "load" js function in inpost webpage
+        location = {
+            "position": {"lat": v["l"]["a"], "lng": v["l"]["o"]},
+            "street": v["e"],
+            "housenumber": v["b"],
+            "province": v["r"],
+            "post_code": v["o"],
+            "city": v["c"],
+            "type": v["t"],
+            "name": v["d"],  # originally, location_description
+            "hours": v["h"],
+            "payment": v["p"],
+            "ref": v["n"],  # originally, name
+            "status": v["s"],
+            "apm_doubled": v["m"],
+        }
+        v.clear()
+        v.update(location)
+        v["active"] = int(v["status"]) == 1
+        v["category"] = Categories.PARCEL_LOCKER if int(v["type"]) == 1 else Categories.POST_PARTNER
```

Comment:
We really don't need to use DictParser for everything. This would have probably been less code, and easier to follow if it wasn't.

Reply:
I did that for two different reasons actually:
* to have the mappings in one place and not use the single-letter-key scattered around
* to replicate what the load js function does in the webpage, so next time someone wants to check the mapping, it's easier to validate and, in case, update

In doing that I took the occasion to use the DictParser known names to reduce clutter later in the code